torchvision/models/vision_transformer.py

```diff
@@ -284,10 +285,20 @@ def _vision_transformer(
     progress: bool,
     **kwargs: Any,
 ) -> VisionTransformer:
-    image_size = kwargs.pop("image_size", 224)
-
     if weights is not None:
         _ovewrite_named_param(kwargs, "num_classes", len(weights.meta["categories"]))
+        if "size" in weights.meta:
+            if isinstance(weights.meta["size"], int):
+                _ovewrite_named_param(kwargs, "image_size", weights.meta["size"])
+            elif isinstance(weights.meta["size"], abc.Sequence):
+                torch._assert(
```

Comment:
FYI: Here you can use Exceptions because it executed during the construction of the model. The issue is when Exceptions appear in the forward. Up to you if you want to keep it as is.